skgstat/Kriging.py

```diff
@@ -389,16 +395,23 @@ def _krige(self, idx):
         dists = self.transform_dists[idx,:]
         
         # find all points within the search distance
-        idx = np.where(dists <= self.range)[0]
+        if isinstance(dists, scipy.sparse.spmatrix):
+            idx = np.array([k[1] for k in dists.keys()])
```

Comment:
For non-sparse datasets this might actually be a performance bottleneck, and we should use toarray() or somesuch solution to speed it up.